### ISSUE TITLE
[BUG] Model fitting time increases significantly starting from version 1.16.0

### ISSUE THREAD

**Issue by romqn1999** [PreferredAI/cornac#557]
### Description
I am utilizing the BPR model for my recommender system.

I have experimented with the same class `BPR` across different versions while training the models on identical datasets. The results showed that:
- Using the library version `<=1.15.4`, the process rate is approximately `3it/s`
- With the library version `>=1.16.0`, the process rate is around `100s/it`

### In which platform does it happen?
```bash
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

### How do we replicate the issue?
My script runs in a venv with `python3.9`
```python
from cornac.data import Dataset
from cornac.models import BPR


train_set = Dataset.from_uir(...)

model = BPR(k=128, max_iter=500, learning_rate=0.05, lambda_reg=0.001, verbose=True)

model.fit(train_set)
```

### Expected behavior
The fitting time for the same model in the later versions should not be longer than the fitting time in the earlier versions.

**Comment by tqtg** [comment]
Thanks for raising this. We've identified the issue that it's due to behavior changes of Cython in the recent migration. We're working on a fix (#558). 

Please rest assured that there shouldn't any changes in the performance of BPR model in version `1.15.4` as compared to the latest. You can safely use that version until we incorporate the fix into our next release. In case you want to use certain features only available from version `1.16`, you can pull the above PR and install Cornac from source.

Let us know if this resolves your issue. Thanks.

**Comment by romqn1999** [comment]
Thank you for your prompt reply. I'm considering using the BPR model in version `1.15.4` for now.

**Comment by tqtg** [comment]
The fix is now released in [v1.18](https://github.com/PreferredAI/cornac/releases/tag/v1.18.0). Closing this issue.
